homepage/src/content/blog/ts-enums/typescript-enums.md

```diff
@@ -0,0 +1,133 @@
+to my `HttpStatusCode` Enum, I just picked some common one, but just have a quick look at the [https://en.wikipedia.org/wiki/List_of_HTTP_status_codes](full%20list), and it should give you a rough impression what it does
+to your bundle size.
+There are applications out there, where bundle size is not a concern at all, but I still have a problem with this approach. It breaks the distinct line between TypeScript and JavaScript.
+There is one fix to that problem, `const enums` in TypeScript don't create any code at runtime, they just act as a type and their value is automatically resolved when used.
+
+![Showing the code of a TS Enum defining some HTTP Status Code and also the resulting JS code](./ts-const-enum.png)
+
+[//]: # (https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwgOwM72MgrgWzgCRhjAGUYBDGLVAYQgBNg4BvAKDg7gHkBpOAXjgAmAAwiANO04AhAIIARAPoAlAKIBFAKqqSAFQFwALGMmc4mgHKzNu-F2UBJAFqr5B4wEZTnAGL3pDvLyqhbuIgDM3hwWXLqKfpZugsaGUXAOFrqqylYAMook2QBq2YrZyvYGAKxirAC+rEho8CAGhMRklNR0jAB0vEA)
+
+The result of this looks great to me, but you can run into problems with this.
+It is getting more and more common to use other transpilers to generate JavaScript code from TypeScript.
+Often those transpilers work on a file-by-file basis,
+transpiling just a single file without fully understanding the full type system.
+[Vite](https://vitejs.dev/) and [Babel](https://babeljs.io/) work this way for instance.
+Just transpiling a single file does not read imported modules, so there is no way to support `const enums`.
+
+## The Solution
+
+Well first and foremost don't use enums!
+TypeScript has Literal Types that are a great replacement for enums.
+Literal Types in TypeScript can be used for strings and numbers (and booleans but less relevant for this blog post) and the best thing is: those are type safe.
+Alright, here's how using literal types for our `HttpStatusCode` enum could look like this:
+
+```ts twoslash
+export const HttpStatusCode_OK = 200;
+export const HttpStatusCode_BAD_REQUEST = 400;
+export const HttpStatusCode_UNAUTHORIZED = 401;
+export const HttpStatusCode_FORBIDDEN = 403;
+export const HttpStatusCode_NOT_FOUND = 404;
+export const HttpStatusCode_INTERNAL_SERVER_ERROR = 500;
+
+export const ALL_HTTP_STATUS_CODES = [
+  HttpStatusCode_OK,
+  HttpStatusCode_BAD_REQUEST,
+  HttpStatusCode_UNAUTHORIZED,
+  HttpStatusCode_FORBIDDEN,
+  HttpStatusCode_NOT_FOUND,
+  HttpStatusCode_INTERNAL_SERVER_ERROR,
+] as const;
+
+export type HttpStatusCodes = typeof ALL_HTTP_STATUS_CODES[number]
+```
+
+This is a pattern I mostly use for this purpose.
+It doesn’t support the bidirectional-access of enums, but this is something I rarely use and could be accomplished via copy paste.
+It also works the same way with strings instead of numbers.
+This approach doesn't come with footguns and is pretty straightforward.
+I also like the type safety of it.
```

Comment:
Could you add a comment that shows what `HttpStatusCode` expands to? If might not be obvious that this is essentially a union type.